Sources/Swift/Integrations/SessionReplay/SentryOnDemandReplay.swift

```diff
@@ -21,52 +21,50 @@ enum SentryOnDemandReplayError: Error {
     case cantReadVideoSize
 }
 
-@available(iOS 16.0, tvOS 16.0, *)
 @objcMembers
 class SentryOnDemandReplay: NSObject {
     private let _outputPath: String
-    private let _onDemandDispatchQueue: DispatchQueue
-    
-    private var _starttime = Date()
-    private var _frames = [SentryReplayFrame]()
     private var _currentPixelBuffer: SentryPixelBuffer?
+    private var _totalFrames = 0
     
+    var frames = [SentryReplayFrame]()
+    var workingQueue: DispatchQueue
```

Comment:
`m`: I think it would be better to add a second initializer to keep this private. Now, you could change this after initializing the class while a couple of blocks are already running async, which opens a door for bugs. The same applies to `dateProvider`.